deploy/cdk_exec_policy/cdkExecPolicy.yaml

```diff
@@ -0,0 +1,280 @@
+AWSTemplateFormatVersion: 2010-09-09
+Description: Custom least privilege IAM policy for linking environments to dataall
+Parameters:
+  AwsAccountId:
+    Description: AWS AccountId of the account that we wish to link.
+    Type: String
+  PolicyName:
+    Description: IAM policy name (The same name must be used during CDK bootstrapping. Default is DataAllCustomCDKPolicy.)
+    Type: String
+    Default: 'DataAllCustomCDKPolicy'
+  EnvironmentResourcePrefix:
+    Description: The resource prefix value of the dataall environment. It MUST match the resource prefix that we use when we create the environment.
+    Type: String
+Resources:
+  CDKCustomExecutionPolicy0:
+    Type: 'AWS::IAM::ManagedPolicy'
+    Properties:
+      ManagedPolicyName: !Ref PolicyName
+      PolicyDocument:
+        Version: 2012-10-17
+        Statement:
+          - Sid: KMS
+            Action:
+              - 'kms:CreateKey'
+            Effect: Allow
+            Resource: '*'
+          - Sid: ec2
+            Action:
+              - 'ec2:DescribeSecurityGroups'
+              - 'ec2:CreateSecurityGroup'
+              - 'ec2:createTags'
+              - 'ec2:DeleteSecurityGroup'
+              - 'ec2:RevokeSecurityGroupEgress'
+              - 'ec2:AuthorizeSecurityGroupIngress'
+              - 'ec2:AuthorizeSecurityGroupEgress'
+            Effect: Allow
+            Resource: '*'
+          - Sid: Athena
+            Effect: Allow
+            Action: 'athena:CreateWorkGroup'
+            Resource:
+              - !Sub 'arn:aws:athena:*:${AWS::AccountId}:workgroup/*'
+          - Sid: IAM
+            Action:
+              - 'iam:CreatePolicy'
+              - 'iam:GetPolicy'
+            Effect: Allow
+            Resource:
+              - !Sub 'arn:aws:iam::${AWS::AccountId}:policy/*'
+          - Sid: IAMRole
+            Action:
+              - 'iam:AttachRolePolicy'
+              - 'iam:CreateRole'
+              - 'iam:CreateServiceLinkedRole'
+              - 'iam:GetRole'
+              - 'iam:GetRolePolicy'
+              - 'iam:PutRolePolicy'
+              - 'iam:TagRole'
+              - 'iam:UnTagRole'
+              - 'iam:DeleteRole'
+              - 'iam:DetachRolePolicy'
+              - 'iam:DeleteRolePolicy'
+              - 'iam:PassRole'
+              - 'iam:UpdateAssumeRolePolicy'
+              - 'iam:ListPolicyVersions'
+              - 'iam:DeletePolicy'
+              - 'iam:ListPolicies'
+              - 'iam:ListAttachedRolePolicies'
+              - 'iam:ListGenerateServiceLastAccessedDetails'
+              - 'iam:ListEntitiesForPolicy'
+              - 'iam:ListPoliciesGrantingServiceAccess'
+              - 'iam:ListRoles'
+              - 'iam:GetServiceLastAccessedDetails'
+              - 'iam:ListAccountAliases'
+              - 'iam:ListRolePolicies'
+            Effect: Allow
```

Comment:
Here I think we can add  `iam:List*` and remove each call out of `iam:List...` to consolidate this statement

Reply:
I did both the above changes. Yes we should refine this further to split into multiple policies for the long-term.